### PR TITLE
[SPARK-47040][CONNECT][FOLLOWUP] Improve usability of the start-conect-server-script.sh

### PR DESCRIPTION
**Pull request by grundprinzip**
### What changes were proposed in this pull request?
Changes the script to use `SPARK_NO_DAEMONIZE` to start the connect service.

### Why are the changes needed?
Ok

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
Manual

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

**Review by dongjoon-hyun — DISMISSED** [review]
For this one, Apache Spark community recommends to use `[FOLLOWUP]` tag of the previous JIRA, SPARK-47040, @grundprinzip .

- https://github.com/apache/spark/pull/45090

**Review by hvanhovell — APPROVED** [review]
LGTM

**Comment by dongjoon-hyun** [comment]
Thank you for update, @grundprinzip .

**Review by dongjoon-hyun — APPROVED** [review]

File: sbin/start-connect-server.sh [hunk 1]
```diff
@@ -40,8 +40,9 @@ fi
 
 if [ "$1" == "--wait" ]; then
   shift
-  exec "${SPARK_HOME}"/bin/spark-submit --class $CLASS 1 --name "Spark Connect Server" "$@"
-else
-  exec "${SPARK_HOME}"/sbin/spark-daemon.sh submit $CLASS 1 --name "Spark Connect server" "$@"
+  export SPARK_NO_DAEMONIZE=1
 fi
+  exec "${SPARK_HOME}"/sbin/spark-daemon.sh submit $CLASS 1 --name "Spark Connect server" "$@"
```
Comment (dongjoon-hyun) [inline sbin/start-connect-server.sh:45]: While checking once more, I realized that we have extra leading spaces in this line. Could you double-check the indentation?

Screenshot 2024-02-15 at 23 10 56

Comment (grundprinzip) [reply sbin/start-connect-server.sh:45]: Done.

**Comment by dongjoon-hyun** [comment]
Thank you, @grundprinzip .
Merged to master.

**Pull request closed without merging**